pkg/api/defaults_test.go

```diff
@@ -1632,11 +1563,11 @@ func TestMasterProfileDefaults(t *testing.T) {
 	// this validates default masterProfile configuration
 	mockCS := getMockBaseContainerService("1.13.12")
 	properties := mockCS.Properties
-	properties.OrchestratorProfile.OrchestratorType = Kubernetes
 	properties.OrchestratorProfile.KubernetesConfig.ClusterSubnet = ""
 	properties.OrchestratorProfile.KubernetesConfig.NetworkPlugin = NetworkPluginAzure
 	properties.MasterProfile.AvailabilityProfile = ""
 	properties.MasterProfile.Count = 3
+	properties.CertificateProfile = &CertificateProfile{}
```

Comment:
Why did this need to be added to the test (and below)? I'm curious...

Reply:
I'm not sure it's strictly required. Basically the `certsAlreadyPresent`. While I was massaging some of these UT it seemed preferable to use mock certs to avoid generating them at runtime (making tests slower).

Reply:
Oh nevermind, these were added to reproduce the existing UT behaviors, where the certs *are* generated at runtime for the bulk of tests. Because I updated the mock data structure to include mock certs (which would bypass on-demand cert generation).

So I think in the future we can consider removing these zero value `CertificateProfile` statements to save UT time, and not generate certs everywhere. But I didn't want to completely remove functional unit testing of on-demand cert generation.

Reply:
Makes sense to me.